temporal/t.rast.what/t.rast.what.py

```diff
@@ -460,84 +454,84 @@ def one_point_per_col_output(
+                        if vcat:
+                            cat = row[0]
+                            x = row[1]
+                            y = row[2]
+                            out_str += (
+                                "{sep}{cat}{csep}{x:10.10f}{csep}"
+                                "{y:10.10f}".format(
+                                    cat=cat,
+                                    x=float(x),
+                                    y=float(y),
+                                    sep=separator,
+                                    csep=coor_sep,
+                                )
+                            )
+                            if site_input:
+                                site = row[3]
+                                out_str += "{sep}{site}".format(sep=coor_sep, site=site)
+                        else:
+                            x = row[0]
+                            y = row[1]
+                            out_str += "{sep}{x:10.10f}{csep}{y:10.10f}".format(
+                                x=float(x), y=float(y), sep=separator, csep=coor_sep
+                            )
+                            if site_input:
+                                site = row[2]
+                                out_str += "{sep}{site}".format(sep=coor_sep, site=site)
 
-                out_file.write(out_str + "\n")
+                    out_file.write(out_str + "\n")
 
-        first = False
+            first = False
 
-        if vcat:
-            ncol = 4
-        else:
-            ncol = 3
-        for col in range(num_cols - ncol):
-            start, end = output_time_list[count][col].get_temporal_extent_as_tuple()
-            time_string = "%(start)s%(sep)s%(end)s" % (
-                {"start": str(start), "end": str(end), "sep": separator}
-            )
-            out_file.write(time_string)
-            for row in range(len(matrix)):
-                value = matrix[row][col + ncol]
-                out_file.write(
-                    "%(sep)s%(value)s" % ({"sep": separator, "value": value.strip()})
+            if vcat:
+                ncol = 4
+            else:
+                ncol = 3
+            for col in range(num_cols - ncol):
+                start, end = output_time_list[count][col].get_temporal_extent_as_tuple()
+                time_string = "%(start)s%(sep)s%(end)s" % (
+                    {"start": str(start), "end": str(end), "sep": separator}
                 )
-            out_file.write("\n")
+                out_file.write(time_string)
+                for row in range(len(matrix)):
+                    value = matrix[row][col + ncol]
+                    out_file.write(
+                        "%(sep)s%(value)s"
+                        % ({"sep": separator, "value": value.strip()})
+                    )
+                out_file.write("\n")
 
-        in_file.close()
-    if out_file is not sys.stdout:
-        out_file.close()
+            in_file.close()
```

Comment:
It's kind of hard to review this due to the changed indentation, but shouldn't this be removed?

Reply:
Yes. At first I wasn't as sure, but in the 3 later PRs (not submitted here yet), I've been doing it, after having tried it more.

Unfortunately, there will be a lots of places in future PRs where the indentation will change. Using side by side comparaison instead of inline would be easier to follow. I am trying to avoid just indenting everything. There are simple cases where a Path(a_file_var).read_text() can be used instead, and I try to use it first. 
(See example in https://docs.astral.sh/ruff/rules/read-whole-file/ and https://docs.astral.sh/ruff/rules/write-whole-file/).

Some places I try to just change the order if there are no dependencies so the `with` is on its own without having to change the whole indent. It's difficult when it's used way below afterwards (like stds_export in a future PR, the diff is a little harder to follow).

Reply:
What about refactoring some of the functionality out to functions first so that the investment into the review is for the refactoring rather than just with statement? (Probably as a PR separate from this one.)

Reply:
Also, I'm for applying any "read/write whole file" changes first before this PR.

Reply:
> Also, I'm for applying any "read/write whole file" changes first before this PR.

The ones that could be detected automatically were done in https://github.com/OSGeo/grass/pull/4047 in mid July. The rule is active since so no more new introductions can happen.

It's only after some manual refactorings that I saw new occurrences appear.

Reply:
> What about refactoring some of the functionality out to functions first so that the investment into the review is for the refactoring rather than just with statement? (Probably as a PR separate from this one.) 

You're aware that there are 368+ places where there's missing a context manager for opened files? It's something a bit big... When the goal is to reduce some new Pylint errors, and help introducing pytest on windows....

Reply:
Can we make a plan how to handle all those? It will be a lot to review even if the changes would be trivial.

If the edits are automated and 1:1, then we can say we will review only a sample. What I as suggesting before is that we could treat that based on categories: dev/null group, stdout group, pathlib one-liners, huge changes which may as well be refactoring, ...

Reply:
They are not automated, yet. The patterns are easy to apply, so in theory there is nothing that would prevent ruff to create a fix in the future, buts it's just not done yet.

It ends up being a little keyboard dance, so it's not that long to do.